src/main/java/com/pnu/dev/radioserviceapi/service/VideoServiceImpl.java

```diff
@@ -1,7 +1,125 @@
 package com.pnu.dev.radioserviceapi.service;
 
+import com.pnu.dev.radioserviceapi.client.YoutubeApiClient;
+import com.pnu.dev.radioserviceapi.client.dto.videos.ItemYoutubeVideosResponse;
+import com.pnu.dev.radioserviceapi.exception.RadioServiceAdminException;
+import com.pnu.dev.radioserviceapi.exception.ServiceException;
+import com.pnu.dev.radioserviceapi.mongo.Video;
+import com.pnu.dev.radioserviceapi.repository.VideoRepository;
+import com.pnu.dev.radioserviceapi.util.mapper.VideoMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 @Service
 public class VideoServiceImpl implements VideoService {
+
+    private final VideoRepository videoRepository;
+
+    private final YoutubeApiClient youtubeApiClient;
+
+    private final VideoMapper videoMapper;
+
+    private final static String youtubeLinkRegex =
+            "http(?:s?):\\/\\/(?:www\\.)?youtu(?:be\\.com\\/watch\\?v=|\\.be\\/)([\\w\\-\\_]*)(&(amp;)?\u200C\u200B[\\w\\?\u200C\u200B=]*)?";
+
+    @Autowired
+    public VideoServiceImpl(VideoRepository videoRepository, YoutubeApiClient youtubeApiClient, VideoMapper videoMapper) {
+        this.videoRepository = videoRepository;
+        this.youtubeApiClient = youtubeApiClient;
+        this.videoMapper = videoMapper;
+    }
+
+    @Override
+    public List<Video> findAll() {
+
+        return videoRepository.findAll(Sort.by("priority").ascending());
+    }
+
+    @Override
+    @Transactional
+    public void deleteById(String id) {
+
+        Video video = findVideoByIdOrThrowException(id);
+        changePriorityFor(video.getPriority(), Integer.MAX_VALUE, -1);
+        videoRepository.deleteById(id);
+    }
+
+    @Override
+    public Video findVideoOnYoutube(String link) throws ServiceException {
+
+        Pattern youtubeLinkPattern = Pattern.compile(youtubeLinkRegex);
```

Comment:
Extract this pattern matching to some method with name: `fetchVideoIdFromUri`.
Can be extracted even to separate component.